src/appleDeviceUtils/appleDeviceUtils.test.js

```diff
@@ -0,0 +1,40 @@
+const {wkSendAndWait} = require('./AppleDeviceUtils')
+
+const ddgGlobals = window.navigator.ddgGlobals
+
+const webkitMock = jest.fn(async (data) => {
+    const {messageHandling} = data
+
+    if (messageHandling.secret !== 'PLACEHOLDER_SECRET') return
+
+    const message = {data: 'test'}
+
+    const iv = new ddgGlobals.Uint8Array(messageHandling.iv)
+    const keyBuffer = new ddgGlobals.Uint8Array(messageHandling.key)
+    const key = await ddgGlobals.importKey('raw', keyBuffer, 'AES-GCM', false, ['encrypt'])
+
+    const encrypt = (message) => {
+        let enc = new ddgGlobals.TextEncoder()
+        return ddgGlobals.encrypt(
+            {
+                name: 'AES-GCM',
+                iv
+            },
+            key,
+            enc.encode(message)
+        )
+    }
+
+    encrypt(JSON.stringify(message)).then((encryptedMsg) =>
+        window[messageHandling.methodName](encryptedMsg))
```

Comment:
This will basically be what the native layer injects to send the response, within a self-invoking function.